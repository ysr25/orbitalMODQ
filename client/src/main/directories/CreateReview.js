import React, { Component } from 'react'
import ReviewForm from '../components/ReviewForm'

export default class CreateReview extends Component {
  constructor(props) {
    super(props)

    this.state = {
      title: '',
      content: '',
      moduleCode: '',
      isAnonymous: false,

      status: null,
      isButtonDisabled: false
    }
  }

  onChangeTitle = (data) => {
    this.setState({ 
      title: data,
      isButtonDisabled: false
    })
  }

  onChangeContent = (data) => {
    this.setState({ 
      content: data,
      isButtonDisabled: false
    })
  }

  onChangeModuleCode = (data) => {
    this.setState({ 
      moduleCode: data,
      isButtonDisabled: false
    })
  }

  onChangeIsAnonymous = () => {
    this.setState(prevState => ({ 
      isAnonymous: !prevState.isAnonymous,
      isButtonDisabled: false 
    }))
  }

  onSubmit = (e) => {
    e.preventDefault()

    const newPost = {
      title: this.state.title,
      content: this.state.content,
      moduleCode: this.state.moduleCode,
      anonymous: this.state.isAnonymous
    }

    this.setState({
      status:'Submitting...',
      isButtonDisabled: true
    })

    this.props.api('post', '/reviews', newPost)
      .then(res => {
        this.setState({ status: res.data.message })
        this.props.history.push(`/reviews/view/${res.data.content}`)
      })
      .catch(err => {
        this.setState({ 
          status: err.response.data.message,
          isButtonDisabled: true
        })
      })
  }

  render() {
    return (
      <>
        <h3>New Post</h3>
        <ReviewForm 
          title={this.state.title}
          content={this.state.content}
          moduleCode={this.state.moduleCode}
          isAnonymous={this.state.isAnonymous}

          status={this.state.status}
          isButtonDisabled={this.state.isButtonDisabled}

          onChangeTitle={this.onChangeTitle}
          onChangeContent={this.onChangeContent}
          onChangeModuleCode={this.onChangeModuleCode}
          onChangeIsAnonymous={this.onChangeIsAnonymous}
          onSubmit={this.onSubmit}
        />
      </>
    )
  }
}
