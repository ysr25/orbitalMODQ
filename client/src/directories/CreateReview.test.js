import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import CreateReview from "./CreateReview";

it("renders create module review page", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <CreateReview />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
