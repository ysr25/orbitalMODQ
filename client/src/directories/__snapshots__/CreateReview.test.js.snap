// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders create module review page 1`] = `
Array [
  <h3>
    New Post
  </h3>,
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="title"
      >
        Title
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          id="title"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <br />
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="module"
      >
        Module
      </label>
      <div
        className="col"
      >
        <input
          autoComplete="off"
          className="form-control"
          id="module"
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder="Choose a module"
          value=""
        />
      </div>
    </div>
    <br />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    <br />
    <div
      className="form-check"
    >
      <input
        aria-label="Post Anonymously"
        checked={false}
        className="form-check-input"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        title=""
      >
        Post Anonymously
      </label>
    </div>
    <br />
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
    <br />
  </form>,
]
`;
